### DISCUSSION TITLE
Is there a way to load all previous pages?

### DISCUSSION THREAD

**Discussion by slaetthult** [algolia/instantsearch#6058] [Q&A] [answered]
Is there a way to load all previous pages without the "show previous results" button with infiniteHits? 
I need to show all previous items **initially**.

**Comment by Haroenv** [comment, marked as answer]
No, this could cause a massive amount of requests (which are paid), and would cause issues with billing as an effect. Unfortunately this is the most workable workaround we have.

However the sessionStorageCache should fix simply reloading the page and getting what you've already seen

**Reply by slaetthult** [reply]
Thank you for your support! This should help. 
I just want to note that the documentation how to implement the session storage cache is outdated for InstantSearch.js v4:
https://www.algolia.com/doc/api-reference/widgets/infinite-hits/js/?client=sessionStorage#widget-param-cache

Maybe we can add a cross reference in the documentation for "showprevious".
https://www.algolia.com/doc/api-reference/widgets/infinite-hits/js/#widget-param-showprevious
